ignite/distributed/auto.py

```diff
@@ -266,22 +304,30 @@ class DistributedProxySampler(DistributedSampler):
 
     """
 
-    def __init__(self, sampler: Sampler, num_replicas=None, rank=None):
+    def __init__(self, sampler: Sampler, num_replicas: Optional[int] = None, rank: Optional[int] = None) -> None:
 
         if not isinstance(sampler, Sampler):
             raise TypeError("Argument sampler should be instance of torch Sampler, but given: {}".format(type(sampler)))
 
         if not hasattr(sampler, "__len__"):
             raise TypeError("Argument sampler should have length")
 
-        super(DistributedProxySampler, self).__init__(sampler, num_replicas=num_replicas, rank=rank, shuffle=False)
+        super(DistributedProxySampler, self).__init__(
+            sampler, num_replicas=num_replicas, rank=rank, shuffle=False  # type: ignore[arg-type]
+        )
         self.sampler = sampler
 
-    def __iter__(self):
+    def __setattr__(self, name: str, value: Any) -> None:
```

Comment:
I wonder why we need override `__setattr__` and `__getattr__` here ?

Reply:
It's because of this part here
https://github.com/pytorch/ignite/blob/3f8daafa7ea119bcf2845c7d042d576f2b9ec122/ignite/distributed/auto.py#L110-L120
`_MpDeviceLoader` doesn't have a attribute `sampler`

Reply:
As now the latest XLA includes `MpDeviceLoader`, I think we can rewrite this part of the code. 
Do you know if there is any mypy correct way to add an attribute to an instance ?

Reply:
the only thing I found so far was to use `__setattr__` and  `__getattr__`

Reply:
I will remove them and put `# type: ignore` next to it. This probably done via a separate PR.

Reply:
So, will you remove them here or in another PR ?